Runtime/HighLevel/ScormTracker.cs

```diff
@@ -0,0 +1,337 @@
+        /// <param name="type">Type.</param>
+        public StatementPromise Progressed(string id, ScormType type, float value)
+        {
+            return Enqueue(new Statement
+            {
+                verb = GetVerb(Verb.Progressed),
+                target = GetTargetActivity(id, type),
+                result = new Result {
+                    score = new Score { scaled = value,}
+                },
+                context= GetContext(ContextActivity.Scorm)
+            });
+        }
+        #endregion
+
+        
+        #region Terminated
+        /// <summary>
+        /// Player terminated a lesson.
+        /// Type = SCO 
+        /// </summary>
+        /// <param name="id">AU id.</param>
+        /// <param name="type">AU type.</param>
+        public StatementPromise Terminated(string id,bool hasDuration=false, long durationInSeconds=0)
+        {
+            if (!initializedTimes.ContainsKey(id))
+            {
+                throw new XApiException("The Terminated statement for the specified id has not been initialized!");
+            }
+
+            // Get the initialized statement time to calculate the duration
+            DateTime ticks = initializedTimes[id];
+            initializedTimes.Remove(id);
+
+            TimeSpan duration = hasDuration ? TimeSpan.FromSeconds(durationInSeconds) : DateTime.Now - ticks;
+            return Enqueue(new Statement
+            {
+                verb = GetVerb(Verb.Terminated),
+                target = GetTargetActivity(id, ScormType.SCO),
+                context = GetContext(ContextActivity.Scorm),
+                result= new Result {
+                    duration = duration,
+                }
+            });
+        }
+        #endregion
+
+
+        #region Passed
+
+        /// <summary>
+        /// The learner attempted and succeeded in a judged activity in the AU.
+        /// </summary>
+        /// <param name="id">AU id.</param>
+        /// <param name="type">AU type.</param>
+        /// <param name="score">The score scaled.</param>
+        public StatementPromise Passed(string id, float score, double durationInSeconds)
+        {
+            return Enqueue(new Statement
+            {
+                verb = GetVerb(Verb.Passed),
+                target = GetTargetActivity(id, ScormType.SCO),
+                context =  GetContext(ContextActivity.Scorm),
+                result = new Result
+                {
+                    success = true,
+                    score = float.IsNaN(score) ? null : new Score { scaled = score },
+                    duration = TimeSpan.FromSeconds(durationInSeconds)
+                }
+            });
+        }
+
+        #endregion
+
+        #region Failed
+        /// <summary>
+        /// The learner attempted and failed in a judged activity in the AU.
+        /// The (scaled) score MUST be lower than the "masteryScore"
+        /// indicated in the LMS Launch Data.
```

Comment:
It seems like some of the comments in the file reflect cmi5 terminology (AU o Assignable Units, masteryscore, LMS Launch Data) rather than SCORM .

Please I suggest review the comments before the merge

Reply:
fixed